File: spring-web-java/src/main/java/spring/web/java/infrastructure/config/JpaConfig.java
package spring.web.java.infrastructure.config;

import java.util.Optional;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.domain.AuditorAware;
import org.springframework.data.jpa.repository.config.EnableJpaAuditing;

import spring.web.java.infrastructure.util.SecurityContextUtil;

@EnableJpaAuditing
@Configuration(proxyBeanMethods = false)
public class JpaConfig {

	@Bean
	public AuditorAware<Long> auditorProvider() {
		return () -> {
			try {
				return Optional.of(SecurityContextUtil.getMemberId());
			} catch (Exception e) {
				return Optional.empty();
			}
		};
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/infrastructure/config/JpaConfig.kt
package spring.web.kotlin.infrastructure.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.domain.AuditorAware
import org.springframework.data.jpa.repository.config.EnableJpaAuditing
import spring.web.kotlin.infrastructure.util.SecurityContextUtil
import java.util.*

@EnableJpaAuditing
@Configuration(proxyBeanMethods = false)
class JpaConfig {

    @Bean
    fun auditorProvider() = AuditorAware {
        runCatching { Optional.of(SecurityContextUtil.getMemberId()) }.getOrDefault(Optional.empty())
    }
}